bob-common/src/mapper.rs

```diff
@@ -116,22 +122,24 @@ impl Virtual {
     }
 
     pub fn get_support_nodes(&self, key: BobKey, count: usize) -> Vec<&Node> {
+        debug_assert!(count < self.nodes.len());
         trace!("get target nodes for given key");
         let target_nodes = self.get_target_nodes_for_key(key);
         trace!("extract indexes of target nodes");
-        let mut target_indexes = target_nodes.iter().map(Node::index);
-        let len = target_indexes.size_hint().0;
-        debug!("iterator size lower bound: {}", len);
+        let target_indexes: Vec<_> = target_nodes.iter().map(Node::index).collect();
```

Comment:
`collect` will most likely only slow things down. `target_nodes` is a slice so its iteration is super fast. If you want to optimize, then it is better to remove `target_indexes` variable and use `target_nodes` directly below